userland/libtock/crt1.c

```diff
@@ -15,9 +15,11 @@ extern int main(void);
 __attribute__ ((section(".start"), used))
 __attribute__ ((weak))
 __attribute__ ((noreturn))
+__attribute__ ((naked))
 void _start(void* mem_start __attribute__((unused)),
             void* app_heap_break __attribute__((unused)),
             void* kernel_memory_break __attribute__((unused))) {
+  asm volatile ("mov r9, sp");
```

Comment:
This needs a comment explaining that this assumes the kernel places the GOT right above the stack top